js/src/setup-mc/setup-stepper/setup-free-listings/useAutoSaveShippingRatesEffect.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '.~/data';
+import useIsEqualRefValue from '.~/hooks/useIsEqualRefValue';
+import useDebouncedCallbackEffect from '.~/hooks/useDebouncedCallbackEffect';
+
+const useAutoSaveShippingRatesEffect = ( shippingRates ) => {
+	const { saveShippingRates } = useAppDispatch();
+	const shippingRatesRefValue = useIsEqualRefValue( shippingRates );
+
+	useDebouncedCallbackEffect( shippingRatesRefValue, saveShippingRates );
+};
```

Comment:
I was thinking to make it consistent with others like `useAutoSaveSettingsEffect` and `useAutoClearShippingEffect`, i.e. "auto save" will happen with a debounce delay.

Reply:
Thanks for the explanation!

I thought the only purpose of those debounces was to avoid short intervals between calls to the API, but when that concern doesn't exist, is there a need to maintain consistency for "they are all debounced"? Sorry, I'm not sure what the benefit of this is.

As the result of this implementation is workable, I think this thread is a 💅  thing now.